Formula/v8.rb

```diff
@@ -15,98 +16,213 @@ class V8 < Formula
+  end
+
+  resource "v8-deps-third_party-benchmarks" do
+    url "https://chromium.googlesource.com/v8/deps/third_party/benchmarks.git",
+      :revision => "05d7188267b4560491ff9155c5ee13e207ecd65f"
+  end
+
+  resource "v8-deps-third_party-mozilla--tests" do
+    url "https://chromium.googlesource.com/v8/deps/third_party/mozilla-tests.git",
+      :revision => "f6c578a10ea707b1a8ab0b88943fe5115ce2b9be"
   end
 
   def install
-    # Bully GYP into correctly linking with c++11
-    ENV.cxx11
-    ENV["GYP_DEFINES"] = "clang=1 mac_deployment_target=#{MacOS.version}"
-    # https://code.google.com/p/v8/issues/detail?id=4511#c3
-    ENV.append "GYP_DEFINES", "v8_use_external_startup_data=0"
-
-    # fix up libv8.dylib install_name
-    # https://github.com/Homebrew/homebrew/issues/36571
-    # https://code.google.com/p/v8/issues/detail?id=3871
-    inreplace "tools/gyp/v8.gyp",
-              "'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']",
-              "\\0, 'DYLIB_INSTALL_NAME_BASE': '#{opt_lib}'"
-
-    (buildpath/"build/gyp").install resource("gyp")
-    (buildpath/"third_party/icu").install resource("icu")
-    (buildpath/"buildtools").install resource("buildtools")
-    (buildpath/"base/trace_event/common").install resource("common")
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    output_path = "out.gn/x64.release"
+
+    # Trick gclient to reuse our v8 checkouts
+    base = "gclient-cache/chromium.googlesource.com"
+    (buildpath/"#{base}-v8-v8").install Dir["*"]
+
+    (resources - [resource("depot_tools")]).each do |r|
+      (buildpath/"#{base}-#{r.name}").install r
+    end
+
+    # Install depot_tools in PATH
+    (buildpath/"depot_tools").install resource("depot_tools")
+    ENV.prepend_path "PATH", buildpath/"depot_tools"
+
+    # Prevent from updating depot_tools on every call
+    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
+    ENV["DEPOT_TOOLS_UPDATE"] = "0"
+
+    # Initialize and sync gclient
+    system "gclient", "root"
+    system "gclient", "config", "--spec", <<~EOS
+      solutions = [
+        {
+          "url": "https://chromium.googlesource.com/v8/v8.git",
+          "managed": False,
+          "name": "v8",
+          "deps_file": "DEPS",
+          "custom_deps": {},
+        },
+      ]
+      target_os = [ "mac" ]
+      target_os_only = True
```

Comment:
adding a `cache_dir = "#{HOMEBREW_CACHE}/gclient_cache"` here should tell `gclient` to cache it's git checkouts inside `HOMEBREW_CACHE`, see https://dev.chromium.org/developers/how-tos/depottools#TOC-.gclient-file

Reply:
We'll need a matching `brew` PR to tweak `cleanup` to handle that.